### DISCUSSION TITLE
B35T+ support

### DISCUSSION THREAD

**Discussion by sdegeorgio** [webspiderteam/Bluetooth-DMM-For-Windows#49] [General]
I have an Owon B35T+ which I can connect to DMM for Windows over Bluetooth.  The the Bluetooth logo goes bright and the title shows 'Connected', as shown here, however the display does not appear to give any readings.
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/7649ed4c-d9db-4f35-bccd-67c7d71dc76e)

I have successfully connected read the data over Bluetooth using this python script https://github.com/53845714nF/OWON_B35T

Is there anything I can do to help debug the issue so we can get it working with your app?

**Comment by webspiderteam** [comment]
which version of app did you use? Release versions not supported Owon type yet. I only publihed it ,n this comment https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/discussions/47#discussioncomment-8820484

if you used the version in this comment can you upload log file under app folder(files like date_log.txt). so I can test app and fix if there is any issue.

**Reply by webspiderteam** [reply]
Now I added it as pre-release on releases page. 
https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/releases/tag/WPFReleaseBetaTest1.26

**Reply by sdegeorgio** [reply]
Yes this was the version I tested yielding the result shown in my initial post.  Strangely I have just tested again and instead of saying "BDM" in the title bar it says "Lilliput" this time as shown here:
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/731e0778-3868-4afc-b3a9-3a80901f4407)
Also, I don't see any .log files in the application folder:
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/11998d4f-eba8-4729-bb2c-a8dafc1cb019)
Strangely the "About Software" states I am running version 1.25
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/74efa290-c37f-42f6-87d8-f046137c5760)
And the Settings dialog states your pre-release is available but is not the version I am running:
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/25f6ba56-7edc-4a8c-bf50-1385087c0ff7)

I have tried downloading from your link above, extracting and running three times now so I am definitely running the version you have shared here.

**Reply by webspiderteam** [reply]
I modified code for work only owon. can you test with this 
[Portable.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14791279/Portable.zip)
this should work at least we should have logs

**Reply by sdegeorgio** [reply]
Aah fantastic, this is working.  I'll run through all the functions next week and confirm they are working.  Here are the logs for reference:
[log-2024-03-28.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14792746/log-2024-03-28.txt)

**Reply by webspiderteam** [reply]
Thanks for logs this will help me too much while testing :)  Looks like  device name changed while connection and app didn't realize its owon type DMM. I will add device type selection on connection window and device editing page. Please inform me if there is any wrong display or works as good as it has to be :) thank you for helping and testing... 

BTW there is no remote command like changing mode e.g for owon type yet. I will add them later.

**Reply by webspiderteam** [reply]
I updated and fixed problems with adding device type selection in connection window and device editing page https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/releases/tag/WPFReleaseBetaTest1.26 you can test with this. but probably you have to edit your device properties in setting windows startup page.

**Reply by sdegeorgio** [reply]
Cool thanks, I see the device type selection in the Device Picker window.  I selected B35T+ and connect, and the software connects and works.

**Reply by sdegeorgio** [reply]
I've done some very basic testing:
DC V - Good for positive readings, but not negative, e.g. -0.606 on DVM shows -33.374v on your software
DC mV - Good for positive readings, but not negative, e.g. -107.3 on DVM shows -3384.1v on your software
AC V - Good
AC mV - Good
Ohm - All ranges appear to work.  For open circuit, the DMM displays 0.L M Ohm, however your software freezes its display on the last reading.
Diode - Works with a diode attached but for open circuit DMM display shows .0L V but your software freezes on the last reading.
Continuity beep - Works when something is connected to DMM, but when DMM is open circuit its display shows .0L V and your software freezes on the last reading.
Capacitor - Good
Hz - Good
Duty cycle % - Good, but your software displays the degrees Fahrenheit and the % symbols along with the correct measurement ash shown here:
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/704151d3-a395-46ca-96d1-fabc9ddd5975)
TEMP - Good, although not measured negative temperature.
uA - Positive readings are ok, but negative 7.3 uA shows -3284.1 uA on your software.
mA - Positive readings are ok, but negative 8.80 mA shows -336.48 mA on your software.
A - Positive readings are ok, but negative 0.507 mA shows -33.275 A on your software.

If its of any use, here are the logs from this test session:
[log-2024-04-04.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14865686/log-2024-04-04.txt)

I also ran the meter for about 72 hours collecting volts data from something I'm testing, and I didn't experience any Bluetooth disconnects or software crashes.  The software did appear to become less responsive after this long duration of logging but it was still working normally.

I'd like to run 3x meters together and correlate the resulting data but I couldn't see a timestamp in the .csv output file.  Is this option available, or is the MQTT server the way to go for this.  Do you have any examples of what server software works?

**Reply by webspiderteam** [reply]
Thanks for detailed testing I will work on these issues and I will put here updated version as soon as posible. Logs will help too much. 

> I'd like to run 3x meters together and correlate the resulting data but I couldn't see a timestamp in the .csv output file. Is this option available, or is the MQTT server the way to go for this. Do you have any examples of what server software works?

I didn't add timestamps in app because of longterm chart. when I add timestamps app slows down and causes crashes after 10 hours of work because of memory leak. but with mqtt you can use with timestamp. I added mqtt sample app on my repos. you can try it. If its not enough for you I can make changes on Mqtt test client for your needs..

**Reply by webspiderteam** [reply]
I think I fixed negative value and dutycycle issue but I'm not sure about O.Ls I havent find any O.L data in log but I checked other peoples repos and made some changes. can you check them?
[Portable.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14875625/Portable.zip)

**Reply by sdegeorgio** [reply]
Hey,
All negative readings are now reading correctly :)
Duty cycle now shows just & symbol, i.e. no Fahrenheit symbol
I found a log file is not generated when the DMM display is reading .0L / 0.L on open circuit measurements.  I assume this means the meter just doesn't transmit a value in that state?  That said, the Owon iPhone app shows OL when in this state.
Steph.

**Reply by webspiderteam** [reply]
İs there any error log in app folder other than data log? All resources that I have seen generates O.L from decimal point value. But somehow official app gets it. But there should be data in somewhere. All incoming data should be logged too.

**Reply by sdegeorgio** [reply]
Here is the file list:
![image](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/assets/24863816/b60cad41-3b40-4932-948b-0010aa45b844)

Is uuids.txt useful?
[uuids.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14883087/uuids.txt)
[log-2024-04-05.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14883088/log-2024-04-05.txt)

**Reply by webspiderteam** [reply]
Okay I want to try something. I want to log data before decode. maybe there is an error that we can't see. can you try this if this is logging anything while it shows O.L
[Portable.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14883134/Portable.zip)

**Reply by webspiderteam** [reply]
I found that page using same protocol also If it shows O.L or overflow on it this means my decoder gives error when O.L 
https://pbrunot.github.io/owonb41t/multimeter.html
this is another web app https://martmet.github.io/OW18B/ble

**Reply by ludwich66** [reply]
Oh, something is happening here again!! - That's great!
Which protocol is better/easier/safer in your opinion?
(I haven't seen anything about the OWON yet)
OWON 18B and ANENG 9002 are now almost the same price.
greetings from Munich
ludwich

**Reply by webspiderteam** [reply]
Owons protocols are more understandable and easy to decode new models only has 6 byte data old types has 16 byte. I don't know about other things but connection is same. Decoding easier. But I don't know anything about Accuracy, stability and functionality comparison.

**Reply by sdegeorgio** [reply]
Sorry for the delay getting back to you, I've been away for a week...  Here is the log for resistance measurement mode where '0.L' is displayed on the DVM screen.  Then I changed to diode which displays '.0L' and finally beep mode which displays '0L.'.
[log-2024-04-15.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14977574/log-2024-04-15.txt)

**Reply by sdegeorgio** [reply]
> Oh, something is happening here again!! - That's great! Which protocol is better/easier/safer in your opinion? (I haven't seen anything about the OWON yet) OWON 18B and ANENG 9002 are now almost the same price. greetings from Munich ludwich

This guy has some amazing reviews and reviewed the Owon B35T+ and is the reason I went with the Owon https://lygte-info.dk/review/DMMOwon%20B35T%20UK.html

**Reply by webspiderteam** [reply]
> Here is the log for resistance measurement mode where '0.L' is displayed on the DVM screen. Then I changed to diode which displays '.0L' and finally beep mode which displays '0L.'.
> [log-2024-04-15.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14977574/log-2024-04-15.txt)

Is this log without any measurement? only O.L's? if there is no zero or other measurement value then I can fix this issue with this log :)

**Reply by sdegeorgio** [reply]
Yes exactly, I only captured 0Ls.  If it helps here is the log with some more resistance measurements and 0Ls.
[log-2024-04-15.txt](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14977858/log-2024-04-15.txt)

**Reply by webspiderteam** [reply]
OK :) Looks like Problem fixed :) 
[Portable.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/14978119/Portable.zip)

**Reply by sdegeorgio** [reply]
Yep I've tested it and now looking good, well done!
I had a go at using the MQTT interface over the weekend with Mosquitto->MQTTtoPrometheus->Prometheus->Graphana and it is almost working.  The only thing I'm struggling with is configuring the MQTTtoPrometheus regex to extract the device ID from the Topic path, but I'm sure I'll get there with a little more tinkering :)
Thanks for all your hard work on this, its a really cool product.

**Reply by webspiderteam** [reply]
That made me happy :) Atleast its working as  it should with your device :) 
I dont know about prometeus but as I find out there is no device ID available on most mqtt library. Thats why I added "Use Mac Adrees as Topic". that will seperate each devices readings. 

Thank you very much to you for helping me :)
